File: ios/App/App/Contacts.swift
import Foundation
import Capacitor

@objc(Contacts)
public class Contacts : CAPPlugin {
    @objc func getAll(_ call: CAPPluginCall) {
        let contacts = getAllMocked() // TODO: Replace mocked data with real implementation.
        call.success([
            "contacts": contacts
        ])
    }
    
    private func getAllMocked() -> [Any] {
        return [
            [
                "firstName": "Elton",
                "lastName": "Json",
                "phoneNumbers": ["2135551111"],
                "emailAddresses": ["elton@eltonjohn.com"],
            ],
            [
                "firstName": "Freddie",
                "lastName": "Mercury",
                "phoneNumbers": [],
                "emailAddresses": [],
            ],
        ]
    }
}

File: src/data/contacts.ts
import { Plugins } from '@capacitor/core';

const { Contacts } = Plugins;

export interface Contact {
  id: string;
  firstName: string;
  lastName: string;
  phoneNumbers: string[];
  emailAddresses: string[];
}

export const getContacts = async (): Promise<Contact[]> => {
  try {
    const result = await Contacts.getAll();

    return result.contacts;
  } catch (e) {
    console.error(`ERR (${getContacts.name}):`, e);
  }

  return [];
};
